pkg/services/baremetal/client/robot/robot_client.go

```diff
@@ -45,10 +50,34 @@ type Factory interface {
 	NewClient(Credentials) Client
 }
 
+type LoggingTransport struct {
+	http.RoundTripper
+}
+
+var replaceHex = regexp.MustCompile(`0x[0123456789abcdef]+`)
+
+func (lt *LoggingTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+
+	stack := replaceHex.ReplaceAllString(string(debug.Stack()), "0xX")
+
+	log := utils.GetDefaultLogger("debug")
+
+	resp, err = lt.RoundTripper.RoundTrip(req)
+	if err != nil {
+		log.Info("hetzner robot API. Error.", "err", err, "method", req.Method, "url", req.URL, "stack", stack)
+		return resp, err
+	}
+	log.Info("hetzner robot API called.", "statusCode", resp.StatusCode, "method", req.Method, "url", req.URL, "stack", stack)
+	return resp, nil
+}
+
 // NewClient creates new HCloud clients.
 func (f *factory) NewClient(creds Credentials) Client {
+	client := &http.Client{
+		Transport: &LoggingTransport{http.DefaultTransport},
```

Comment:
you can define the logger in this struct LoggingTransport I think! Redefining it would actually be worse than using the other logger I suppose